src/errors.rs

```diff
@@ -727,3 +728,54 @@ pub enum ApiError {
     #[error("Unknown error: {0:?}")]
     Unknown(String),
 }
+
+impl From<reqwest::Error> for DownloadError {
+    fn from(error: reqwest::Error) -> Self {
+        DownloadError::Network(hide_token(error))
+    }
+}
+
+impl From<reqwest::Error> for RequestError {
+    fn from(error: reqwest::Error) -> Self {
+        RequestError::Network(hide_token(error))
+    }
+}
+
+/// Replaces token in the url in the error with `token:redacted` string.
+pub(crate) fn hide_token(mut error: reqwest::Error) -> reqwest::Error {
+    let url = match error.url_mut() {
+        Some(url) => url,
+        None => return error,
+    };
+
+    if let Some(mut segments) = url.path_segments() {
+        // Usually the url looks like "bot<token>/..." or "file/bot<token>/...".
+        let (beginning, segment) = match segments.next() {
+            Some("file") => ("file/", segments.next()),
+            segment => ("", segment),
+        };
+
+        if let Some(token) = segment.and_then(|s| s.strip_prefix("bot")) {
+            // make sure that what we are about to delete looks like a bot token
+            if let Some((id, secret)) = token.split_once(':') {
+                if secret.len() >= 35
```

Comment:
What is 35 here?

Reply:
The lenght of the secret. On all bot tokens I could find it's 35

Reply:
Ok, then consider giving the magic number a name: `const SECRET_LENGTH: usize = 32;` or something like this.

Reply:
Done